lib/parseSty.js

```diff
@@ -0,0 +1,133 @@
+"use strict";
+
+var balanced = require("balanced-match");
+
+var declarationCommands = ["\\newcommand",
+                           "\\renewcommand",
+                           "\\DeclareRobustCommand",
+                           "\\defSpecFun"];
+
+// Returns a two dimensional array, where each element contains the body of a bracket section and the type of bracket
+// that surrounds it. Makes use of the npm balanced-match module.
+// E.g.: [ [ "in brackets", "[" ], [ "in braces", "{" ] ]
+var findBracketSections = module.exports.findBracketSections = function(string, sections) {
+    if (!sections) {
+        sections = [];
+    }
+    var bracket = balanced("[", "]", string);
+    var brace = balanced("{", "}", string);
+    if (brace && (!bracket || brace.start < bracket.start)) {
+        sections.push([brace.body, "{"]);
+        return findBracketSections(brace.post, sections);
+    }
+    else if (bracket) {
+        sections.push([bracket.body, "["]);
+        return findBracketSections(bracket.post, sections);
+    }
+    return sections;
+};
+
+// Finds the first occurrence in the provided string of the earliest occurring item in the given array, starting from
```

Comment:
I'd call the *string*  *search* or something. 
Indicate that subs is an array 
http://usejsdoc.org/about-block-inline-tags.html